src/libAtomVM/memory.c

```diff
@@ -74,7 +74,7 @@ enum MemoryGCResult memory_ensure_free_with_roots(Context *c, uint32_t size, int
             if (new_free_space > maximum_free_space) {
                 size_t new_memory_size = context_memory_size(c);
                 size_t new_requested_size = (new_memory_size - new_free_space) + maximum_free_space;
-                if (!c->has_min_heap_size || (c->min_heap_size < new_requested_size)) {
+                if (!c->has_min_heap_size || ((size_t) c->min_heap_size < new_requested_size)) {
```

Comment:
I'm not a big fan of casts, since they may hide errors in case any change is done.
I'll rather try to fix types in declaration in a meaningful way.

Reply:
Fixed